### ISSUE TITLE
Add (multiple) accounts using API/Etc.. ?

### ISSUE THREAD

**Issue by Yamakasi** [jasonmunro/cypht#407]
I'm looking into a way to add accounts for a user so he doesn't need to add them himself.

Now I login using API against LDAP which adds the account perfectly but I'm in need of adding multiple using LDAP AUTH or a request/DB insert for Cypht.

For Dovecot I was thinking of using a master user for multi account access which seems to do the job for it.

Any idea ?

**Comment by jasonmunro** [comment]
This is a complicated request and maybe outside the scope of a webmail project? I understand the need but not sure it's the right place for this kind of functionality.

**Comment by Yamakasi** [comment]
> 
> 
> This is a complicated request and maybe outside the scope of a webmail project? I understand the need but not sure it's the right place for this kind of functionality.

I'm trying to find a decent way from a middleware point of view. Not that Cypht has lack on it but more to see if there are standards.

Extra accounts adding could be actually an automated ways, API end point or even direct SQL inserts, like the user does it by himself in Cypth. Issue there is the serialization of settings and such. Can't we "outscope" that process to a simple API ?

Just thinking loudly :)

**Comment by marclaporte** [comment]
@Yamakasi which mail servers do you use?

**Comment by Yamakasi** [comment]
> 
> 
> @Yamakasi which mail servers do you use?

I suppose you mean IMAP servers ?

Storage: Dovecot
MTA: Postfix

I can do do some magic with SQL if needed.

What are you suggesting then ?

**Comment by marclaporte** [comment]
I have no clear plan yet but just some ideas. I would like ad hoc creation of email addresses. Ex.: A team member starts a new project and creates a mailbox and sets permissions for it.

I have no idea how we can make this generic. I suspect it will end up being a specific process for each server.

**Comment by Yamakasi** [comment]
> 
> 
> I have no clear plan yet but just some ideas. I would like ad hoc creation of email addresses. Ex.: A team member starts a new project and creates a mailbox and sets permissions for it.
> 
> I have no idea how we can make this generic. I suspect it will end up being a specific process for each server.

Yes sounds like custom. That's why I don't like glue things together as in your wiki, I write such things from scratch, webmail is just a client.

**Comment by marclaporte** [comment]
Related: https://github.com/jasonmunro/cypht/issues/470

**Comment by marclaporte** [comment]
Closing this. Import tool has been done: https://github.com/cypht-org/cypht/pull/976

And discussions on "Declarative config for multiple accounts" continues here: https://github.com/cypht-org/cypht/issues/974
